### PR TITLE
Migrate to circle2

### PR DESCRIPTION
**Pull request by ale7714**
Old branches will need to cherry pick this commit

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+jobs:
+  test:
+    machine:
+      docker_layer_caching: true
+    working_directory: ~/codeclimate/codeclimate-rubocop
+    steps:
+      - checkout
+      - run: make image
+      - run: make citest
```
Comment (maxjacobson) [inline .circleci/config.yml]: per the failing build, it looks like this should be `make test`

**Comment by ale7714** [comment]
Sorry this is WIP, requested review too eagerly. Wrapping up one more thing. And yes to both things

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - run: make test
+
+  release_images:
+    machine:
+      docker_layer_caching: true
+    working_directory: ~/codeclimate/codeclimate-rubocop
+    steps:
+      - checkout
+      - run:
+          name: Validate owner
```
Comment (ale7714) [inline .circleci/config.yml]: circle 1 has this nice feature https://github.com/codeclimate/codeclimate-rubocop/blob/master/circle.yml#L24. 

Unfortunately not available in circle2 so we're the validation within the job itself. 

`circleci step halt` is a nifty trick to exit the job without failing the build. I learn it from this [post](https://discuss.circleci.com/t/useful-tips-and-best-practices-when-migrating-to-circleci-2-0/23075).

Comment (maxjacobson) [reply .circleci/config.yml]: That is neat!

**Pull request merged**